server/svix-server/src/core/permissions.rs

```diff
@@ -157,3 +157,36 @@ impl FromRequestParts<AppState> for ApplicationWithMetadata {
 struct ApplicationPathParams {
     app_id: ApplicationIdOrUid,
 }
+
+pub enum EventTypeFeatureFlagPermissions {
```

Comment:
This looks overly specific and verbose. I don't like the idea of having a specific permission extractor for whatever function we want, and I like it even less having it in permissions.rs rather next to the rest of the code.

Can't we just also extract the JWT extra info and pass it to the function that uses it?

Reply:
I see you put it in the Permissions struct, shouldn't we just use that directly?

Reply:
The answer to this depends on how we want to do permission checks with axum extractors. Currently, we use this "mixin" style, where the different permission levels are different structs, see `permissions::{ReadAll, Application, Organization}`, these are higher level axum extractors that build on top of `Permissions`.

Reply:
Can't the application extractor also return the feature flags when exist (for example)?

Reply:
Is it ok if I just drop the `FromRequestParts` impl from `EventTypeFeatureFlagPermissions` and put it as a field on `Application`? Because simply putting `EventTypeFeatureFlagSet` by itself into `Application` is not enough. We need to know if this `Application` authorization came from an organization token or an app token, otherwise in the handler we can't differentiate and allow organization tokens access regardless of flags.

Reply:
I don't really remember how the permissions are structured, though sure, it sounds reasonable.

Reply:
Done, now it's just an enum on `ReadAll` (for some reason I was mixing up `Application` and `ReadAll`)